src/overlays/actors/ovl_Bg_Mori_Bigst/z_bg_mori_bigst.c

```diff
@@ -20,39 +36,219 @@ const ActorInit Bg_Mori_Bigst_InitVars = {
+    }
+}
+
+void BgMoriBigst_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgMoriBigst* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx2, &globalCtx2->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void BgMoriBigst_SetupObjectCheck(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_ObjectCheck);
+}
+
+void BgMoriBigst_ObjectCheck(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objBankIndex)) {
+        thisx->draw = BgMoriBigst_Draw;
+        if (Flags_GetClear(globalCtx, thisx->room) && (PLAYER->actor.posRot.pos.y > 700.0f)) {
+            if (Flags_GetSwitch(globalCtx, (thisx->params >> 8) & 0x3F)) {
+                BgMoriBigst_FinalNoop(this, globalCtx);
+            } else {
+                BgMoriBigst_SpawnStalfos(this, globalCtx);
+            }
+        } else {
+            BgMoriBigst_InitialNoop(this, globalCtx);
+        }
+    }
+}
+
+void BgMoriBigst_InitialNoop(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetActionFunction(this, NULL);
+}
+
+void BgMoriBigst_SpawnStalfos(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor* stalfos;
+
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_StalfosWait);
+    Flags_UnsetClear(globalCtx, this->dyna.actor.room);
+    stalfos = Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, ACTOR_EN_TEST, 209.0f, 827.0f,
+                                 -3320.0f, 0, 0, 0, 1);
+    if (stalfos != NULL) {
+        this->dyna.actor.child = NULL;
+        this->dyna.actor.initPosRot.rot.z++;
+    } else {
+        // Second Stalfos failure
+        osSyncPrintf("Warning : 第２スタルフォス発生失敗\n");
+    }
+    Flags_SetClear(globalCtx, this->dyna.actor.room);
+}
+
+void BgMoriBigst_StalfosWait(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((this->dyna.actor.initPosRot.rot.z == 0) &&
+        ((this->dyna.actor.initPosRot.pos.y - 5.0f) <= PLAYER->actor.posRot.pos.y)) {
+        BgMoriBigst_SetupFall(this, globalCtx);
+        func_800800F8(globalCtx, 0xC94, 0x48, &this->dyna.actor, 0);
+    }
+}
+
+void BgMoriBigst_SetupFall(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_Fall);
+}
+
+void BgMoriBigst_Fall(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor_MoveForward(&this->dyna.actor);
+    if (this->dyna.actor.posRot.pos.y <= this->dyna.actor.initPosRot.pos.y) {
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y;
+        BgMoriBigst_Quake(this, globalCtx);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_STONE_BOUND);
+        func_800800F8(globalCtx, 0x3FC, 8, &this->dyna.actor, 0);
+        func_8002DF38(globalCtx, NULL, 0x3C);
+    }
+}
+
+void BgMoriBigst_Quake(BgMoriBigst* this, GlobalContext* globalCtx) {
```

Comment:
I think the main thing this function does is setting action `BgMoriBigst_SetupSpawnStalfosPair`, not shaking the camera, so maybe `BgMoriBigst_Land`?

Since `BgMoriBigst_SetupSpawnStalfosPair` waits and isn't actually a Setup function (which must do things immediately imo) it could be renamed to `BgMoriBigst_Landing` and `BgMoriBigst_Quake` to `BgMoriBigst_SetupLanding`? I think that's the intended effect with the wait after the fall, wait for things to "stabilize"

Reply:
That makes sense.